models/app.models.js

```diff
@@ -37,4 +37,43 @@ exports.fetchArticles = () =>{
     .then(({rows})=>{
         return rows
     })
+}
+
+exports.fetchCommentsByArtId = (article_id)=>{
+    const articleString = `
+    SELECT * FROM articles
+    WHERE article_id = $1;
+`;
+return db
+.query(articleString, [article_id])
+.then(({ rows }) => {
+  if (rows.length === 0) {
+    return Promise.reject({
+      status: 404,
+      message: "This article does not exist",
+    });
+  } else {
+    return rows[0];
+  }
+})
+.then((article) => {
+  const commentsFromDb = `
+        SELECT * FROM comments
+        WHERE article_id = $1
+        ORDER BY created_at DESC;
+    `;
+  return db
+    .query(commentsFromDb, [article.article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return [
+          {
+            message: "There are no comments associated with this article"
+          },
+        ];
+      } else {
+        return rows;
+      }
+    });
+});
```

Comment:
I like what you're thinking here with checking the article exists. We already have a model that gets the article by its id so we can handle this logic using a Promise.all in the controller to check the article exists and update it at the same time.